src/core/include/openvino/core/validation_util.hpp

```diff
@@ -180,13 +180,4 @@ OPENVINO_API bool has_no_labels(const TensorLabel& labels);
 OPENVINO_DEPRECATED("This function is deprecated and will be moved to dev api in 2024.0 release.")
 OPENVINO_API std::vector<PartialShape> get_node_input_partial_shapes(const ov::Node& node);
 
-/// \brief Check if rank is compatible to any of rank from container.
-///
-/// \param rank   Rank to check.
-/// \param ranks  VEctor of ranks used to check input rank compatibility.
-///
-/// \return True if rank compatible to any from ranks, otherwise false.
-OPENVINO_DEPRECATED("This function is deprecated and will be moved to dev api in 2024.0 release.")
-OPENVINO_API bool is_rank_compatible_any_of(const ov::Rank& rank, const std::vector<ov::Rank>& ranks);
```

Comment:
Not sure we can delete public API before deprecation period, especially for LTS
@ilya-lavrenov

Reply:
I can restore it, but deleted it as it was created not so long ago, and I think is used inside operators.
Or merge it after LTS release, as this change is not critical for 2023.3?

Reply:
Ok, let's merge